### ISSUE TITLE
Spain canary Islands

### ISSUE THREAD

**Issue by strassburger** [tmrowco/electricitymap#521]
Spain only Mainland (peninsular) and Baleares are connected to the the european  net.
Data see here:
http://ree.es/en/activities/realtime-demand-and-generation

Canary Islands nets:
http://ree.es/en/activities/canary-islands-electricity-system/canary-electricity-demand-in-real-time

data example: of El Hierro
https://demanda.ree.es/movil/canarias/el_hierro/tablas/1

**Comment by strassburger** [comment]
sorry tipo:
data example: of El Hierro
https://demanda.ree.es/movil/canarias/el_hierro/tablas/2

**Comment by blackleg** [comment]
@corradio Can parser use selenium with phantomjs to scrape data? 

The website that contains data from Canary Islands and Balearic Islands use a angular application and required execute the javascript in a browser to show data.

**Comment by corradio** [comment]
Mmh if it works in python it could.
I haven't checked the website but before starting we should be sure that we have access to the generation mix in real-time (just to avoid working on something we can't use in the end)

**Comment by blackleg** [comment]
We add this [pull request 620](https://github.com/tmrowco/electricitymap/pull/620) to split Spain in electricity subsystems

**Comment by blackleg** [comment]
I started a [python package project](https://github.com/blackleg/reescraper) to scrape data from the REE website. 
To work it require the phantomjs binary are available in your system path.

**Comment by corradio** [comment]
Hmm seems like a simple regexp would enable us to extract a JSON from https://demanda.ree.es/WSvisionaMovilesCanariasRest/resources/demandaGeneracionCanarias?callback=angular.callbacks._2&curva=GCANARIA&fecha=2017-07-12
Wouldn't that be sufficient? It would definitely be better than run a full phantomjs binary.

**Comment by blackleg** [comment]
It's a better solution than phantomjs. I study it.

**Comment by strassburger** [comment]
Thank you for your work.
Please check if it is possible to integrate "El Hierro" at his own.
They have a system called "Gorona del Viento" with 100% RE running right now.
data example: of El Hierro
https://demanda.ree.es/movil/canarias/el_hierro/tablas/2

**Comment by blackleg** [comment]
Canary Islands parser: [Pull 662](https://github.com/tmrowco/electricitymap/pull/622)

**Comment by blackleg** [comment]
@strassburger I can get data by island, but I don't know if is possible to split map by island.

**Comment by corradio** [comment]
@strassburger maybe we should start another issue for this?
In order to finish this one, we still need access to the exchanges to the mainland. Do we have access to the data somewhere?

**Comment by strassburger** [comment]
Sorry, got problems with my english:
I understand the question is if the data we can get from other sources then ree.es.
The answer to this is: no
REE.es is the only source. Only they are acting in the borders of the "ley de transparencia" and are publishing data.
"Gorona del Viento"  en El Hierro do not publish facts at all

**Comment by corradio** [comment]
@strassburger what I meant is that we should create another "github issue" (https://github.com/tmrowco/electricitymap/issues/new) in order to coordinate the work to do every island.

Second question: It would be nice to get some data about how much power is imported/exported between the mainland and the islands. Then we can show it on the map too!

**Comment by strassburger** [comment]
i am afraid this will not help:
Without movil but with https and flash player
https://demanda.ree.es/visionaCan/VisionaHierro.html

**Comment by strassburger** [comment]
Think i got it now:
- The see is very deep , that is the problem:
No interconnection abroad between mainland and canary island-
All islands are are isolated- One exception:
"two of which - Lanzarote and Fuerteventura - are interconnected by a submarine link"
source: 
http://ree.es/en/activities/canary-islands-electricity-system/canary-electricity-demand-in-real-time
Hope it helps

**Comment by corradio** [comment]
OK thanks! So I think we can close this issue once the PR is merged, as the canary islands are independent.
